assets/src/edit-story/components/dropTargets/provider.js

```diff
@@ -174,26 +176,45 @@ function DropTargetsProvider({ children }) {
           resource,
         });
       }
-      combineElements(combineArgs);
-
-      // Reset styles on visisble elements
-      elements
-        .filter(({ id }) => !(id in Object.keys(dropTargets)) && id !== selfId)
-        .forEach((el) => {
-          pushTransform(el.id, {
-            dropTargets: {
-              active: false,
-              replacement: null,
+      const finalizeDrop = (baseColor = null) => {
+        if (baseColor) {
+          combineArgs.firstElement = {
+            ...combineArgs.firstElement,
+            resource: {
+              ...combineArgs.firstElement.resource,
+              baseColor,
             },
+          };
```

Comment:
You could probably also just do `combineArgs.firstElement.resource.baseColor = baseColor`, as `resource` has to exist on the existing `firstElement`, right?